src/utils/generateDateString.ts

```diff
@@ -0,0 +1,25 @@
+import { IDateObject } from '../interfaces/searchPayload.interface';
+import { isEmpty } from './isEmpty';
+
+const generateDateString = (dateObject: IDateObject, isToDate = false): string => {
+  const currentDate = new Date();
+  const month = !isEmpty(dateObject.month) && !isNaN(dateObject.month!) ? dateObject.month! : 1;
+  let day: number;
+  if (isToDate && (isEmpty(dateObject.day) || isNaN(dateObject.day!))) {
+    if (month === currentDate.getMonth() + 1) {
```

Comment:
you should also be checking for the year is present year.
eg: 
day - 19
month - empty
year - 2024
then the end date should be 16-02-2024
but looks like this code returns current day for any feb month irrespective of the year.

Reply:
Thanks for pointing this out. Now it has been addressed.